### ISSUE TITLE
Is there a way to access metrics from training (e.g. loss over epochs)?

### ISSUE THREAD

**Issue by jamestwebber** [YosefLab/scvi-tools#825]
Hey there, just started trying out the new interface and it's very clean, nice work.

One thing I was looking for and didn't see: is there any way to inspect the loss curves after training a model? It looks like `scVI.train()` will always train for 400 epochs and I'm trying to figure out if that's reaching convergence in my case, but I can't seem to find the full ELBO history for the trainer. It would be useful to know if it's basically flat by the end of training or if it's jumping around.

I'm trying to evaluate scVI model estimates of mean expression and found that training the same model architecture multiple times was not quite as stable as I'd expect, so I'm wondering if training is not complete.

**Describe the solution you'd like**
Maybe this information is stored somewhere and I just didn't see it, but if not it seems easy enough to store train and validation metrics in a list or array in the trainer.

**Comment by adamgayoso** [comment]
Hi @jamestwebber, first you need to set the frequency parameter in the train method, (which @galenxing is confusing as None default, maybe should be 0?, or the docstring should say what happens with None.). Then you should be able to access with [`model.history` ](https://www.scvi-tools.org/en/0.7.0-beta.0/api/reference/scvi.model.SCVI.history.html#scvi.model.SCVI.history)

**Comment by jamestwebber** [comment]
Ahh I see, I just didn't figure that out from the docs. Thanks!

**Comment by galenxing** [comment]
Yea that is pretty confusing. I'll change it.

**Comment by jamestwebber** [comment]
For what it's worth I think default of `None` is fine–I read it as "our default setting". My confusion came from the description of the `frequency` parameter: `Frequency with which metrics are computed on the data for train/test/val sets.`

There's nothing in that description that I read as "store the results for inspection later". I interpreted it as something to do with your training procedure and left it at the default. Presumably you compute the ELBO all the time for training purposes, and I wasn't going to mess that up.

So, a better docstring would have cleared that up for me. It would also be nice to inherit or repeat docstrings instead of just using `kwargs`, as it becomes an adventure in the API to figure out what the options are for something like `scVI.train`, which passes `kwargs` to `UnsupervisedTrainer` which passes `kwargs` to `Trainer`...

**Comment by adamgayoso** [comment]
> So, a better docstring would have cleared that up for me. It would also be nice to inherit or repeat docstrings instead of just using kwargs, as it becomes an adventure in the API to figure out what the options are for something like scVI.train, which passes kwargs to UnsupervisedTrainer which passes kwargs to Trainer...

I agree with this, we are working on it...

**Comment by adamgayoso** [comment]
The new version `0.9.0` will have a new `check_val_every_n_epochs` argument (inherited from pytorch lightning) to make it more clear.
